src/formatting/formatter/tally.rs

```diff
@@ -0,0 +1,583 @@
+use std::str::FromStr;
+
+use super::*;
+
+#[derive(Debug)]
+enum Style {
+    ChineseCountingRods,
+    ChineseTally,
+    WesternTally,
+    WesternTallyUngrouped,
+}
+
+impl FromStr for Style {
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Self> {
+        match s {
+            "chinese_counting_rods" => Ok(Style::ChineseCountingRods),
+            "chinese_tally" => Ok(Style::ChineseTally),
+            "western_tally" => Ok(Style::WesternTally),
+            "western_tally_ungrouped" => Ok(Style::WesternTallyUngrouped),
+            x => Err(Error::new(format!("Unknown Style: '{x}'"))),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct TallyFormatter {
+    style: Style,
+}
+
+impl TallyFormatter {
+    pub(super) fn from_args(args: &[Arg]) -> Result<Self> {
+        let mut style = Style::WesternTally;
+        for arg in args {
+            match arg.key {
+                "style" | "s" => {
+                    style = arg.val.parse()?;
+                }
+                other => {
+                    return Err(Error::new(format!(
+                        "Unknown argument for 'rods': '{other}'"
+                    )));
+                }
+            }
+        }
+        Ok(Self { style })
+    }
+}
+
+const HORIZONTAL_CHINESE_COUNTING_RODS_CHARS: [char; 10] =
+    ['〇', '𝍠', '𝍡', '𝍢', '𝍣', '𝍤', '𝍥', '𝍦', '𝍧', '𝍨'];
+
+const VERTICAL_CHINESE_COUNTING_RODS_CHARS: [char; 10] =
+    ['〇', '𝍩', '𝍪', '𝍫', '𝍬', '𝍭', '𝍮', '𝍯', '𝍰', '𝍱'];
+
+const CHINESE_TALLY_CHARS: [char; 5] = ['𝍲', '𝍳', '𝍴', '𝍵', '𝍶'];
+
+impl Formatter for TallyFormatter {
+    fn format(&self, val: &Value, _config: &SharedConfig) -> Result<String, FormatError> {
+        match val {
+            Value::Number { val, .. } => {
```

Comment:
Should we match to unit None or support unit conversions?

Reply:
@MaxVerevkin what do you think about the units? If we do support units should we just add a argument `repr`|`representation` defaulting to "arabic" with these additional formats?

Reply:
I think it makes sense to only support `None` unit. Then if someone provides a use-case for supporting other units, we can do something about it.